cmd/instance_delete.go

```diff
@@ -36,33 +35,64 @@ func (c *instanceDeleteCmd) cmdPreRun(cmd *cobra.Command, args []string) error {
 }
 
 func (c *instanceDeleteCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, c.Zone))
+	ctx := gContext
+	client, err := switchClientZoneV3(
+		ctx,
+		globalstate.EgoscaleV3Client,
+		v3.ZoneName(c.Zone),
+	)
+	if err != nil {
+		return err
+	}
 
-	instance, err := globalstate.EgoscaleClient.FindInstance(ctx, c.Zone, c.Instance)
+	instances, err := client.ListInstances(ctx)
 	if err != nil {
-		if errors.Is(err, exoapi.ErrNotFound) {
-			return fmt.Errorf("resource not found in zone %q", c.Zone)
-		}
 		return err
 	}
 
-	if !c.Force {
-		if !askQuestion(fmt.Sprintf("Are you sure you want to delete instance %q?", c.Instance)) {
-			return nil
+	instanceToDelete := []v3.UUID{}
+	for _, i := range c.Instances {
+		instance, err := instances.FindListInstancesResponseInstances(i)
+		if err != nil {
+			if !c.Force {
+				return err
+			}
+
+			continue
 		}
+
+		if !c.Force {
+			if !askQuestion(fmt.Sprintf("Are you sure you want to delete instance %q?", i)) {
+				return nil
+			}
+		}
+
+		instanceToDelete = append(instanceToDelete, instance.ID)
+	}
+
+	var fns []func() error
+	for _, i := range instanceToDelete {
+		fns = append(fns, func() error {
+			op, err := client.DeleteInstance(ctx, i)
+			if err != nil {
+				return err
+			}
+			_, err = client.Wait(ctx, op, v3.OperationStateSuccess)
+			return err
+		})
 	}
 
-	decorateAsyncOperation(fmt.Sprintf("Deleting instance %q...", c.Instance), func() {
-		err = globalstate.EgoscaleClient.DeleteInstance(ctx, c.Zone, instance)
-	})
+	err = decorateAsyncOperations(fmt.Sprintf("Deleting instance %q...", strings.Join(c.Instances, ", ")), fns...)
 	if err != nil {
 		return err
 	}
 
-	instanceDir := path.Join(globalstate.ConfigFolder, "instances", *instance.ID)
-	if _, err := os.Stat(instanceDir); !os.IsNotExist(err) {
-		if err := os.RemoveAll(instanceDir); err != nil {
-			return fmt.Errorf("error deleting instance directory: %w", err)
+	for _, i := range instanceToDelete {
+		instanceDir := path.Join(globalstate.ConfigFolder, "instances", i.String())
+		if _, err := os.Stat(instanceDir); !os.IsNotExist(err) {
+			if err := os.RemoveAll(instanceDir); err != nil {
+				return fmt.Errorf("error deleting instance directory: %w", err)
+			}
```

Comment:
What does this section do? Might we worth a commend.

Reply:
There is a mechanism at instance creation to create default generated sshkeys (private key stored in dedicated dir and public key registered to the instance)
https://github.com/exoscale/cli/blob/master/cmd/instance_create.go#L220

That let you ssh to it if you did not provide a sshkey.

The big benefit of that, is you can directly use the command `exo compute instance ssh <instance name/id>` and it will do the magic for you

Reply:
and the relation to this section is that it cleans up the generated keys, I suppose.
you could write a comment in the code with this info.